zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/RestoreFromBackupTool.java

```diff
@@ -618,4 +658,26 @@ private void copyProcessedRestoredFilesToDestination(FileTxnSnapLog restoreTempS
       Files.copy(processedFile.toPath(), new File(finalDestinationBase, fileName).toPath());
     }
   }
+
+  /**
+   * If the CLI command has specified a znode path to perform spot restoration,
+   * run the spot restoration tool on that path
+   * @throws IOException
+   */
+  @VisibleForTesting
+  protected void performSpotRestorationIfConfigured() throws IOException {
+    if (znodePathToRestore != null) {
+      if (zkServerConnectionStr == null) {
+        throw new IllegalArgumentException(
+            "ZK server connection info is not provided. Could not perform spot restoration.");
+      }
+      zk = new ZooKeeper(zkServerConnectionStr, CONNECTION_TIMEOUT, (event) -> {
+        System.out.println("WATCHER::" + event.toString());
+      });
+      //snapLog.getDataDir() has "/version-2" at the end of the path, remove "/version-2" by using getParent()
+      spotRestorationTool = new SpotRestorationTool(new File(snapLog.getDataDir().getParent()), zk,
+          znodePathToRestore, restoreRecursively);
+      spotRestorationTool.run();
+    }
```

Comment:
is `zk` shut down anywhere?

Reply:
Good catch. `zk` is not shut down. I added to the SpotRestorationTool.